src/cryptonote_core/blockchain.cpp

```diff
@@ -3253,15 +3258,15 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
           continue;
         }
 
-        const auto existing_quorum = m_service_node_list.get_testing_quorum(service_nodes::quorum_type::obligations, existing_state_change.block_height);
-        if (!existing_quorum)
-        {
-          MERROR_VER("could not get obligations quorum for recent state change tx");
+        crypto::public_key existing_state_change_service_node_pubkey;
+        if (!m_service_node_list.try_resolve_pubkey_in_quorum(quorum_type,
+                                                              service_nodes::quorum_group::worker,
+                                                              existing_state_change.block_height,
+                                                              existing_state_change.service_node_index,
+                                                              existing_state_change_service_node_pubkey))
           continue;
```

Comment:
This is for getting the quorums for transactions state changes that have already occurred, doesn't necessarily mean the incoming TX is invalid, just that the daemon for whatever reason doesn't have access to the quorum for this historical transaction.

Actually this check should also be checking the state_change.state == existing_state_change.state as well. Hmm, but then we'd run into problems with old states rejecting new states prematurely if they experience say, decommission, recommission, decommission, the last decommission would be rejected, even if they were meant to decommission because they were not functioning properly after the recommission.

Reply:
Maybe that's acceptable and we just reject the duplicated state changes, even if they are valid until the window has passed and they'd be resubmitted after the fact.

Reply:
I don't think `state_change.state == existing_state_change.state` will do anything because it shouldn't be possible to get conflicting votes at the same block height (which this is already specific to).

Reply:
Oh right, I was thinking of the check_tx_inputs which goes through the historical blocks, where we have this problem

Reply:
I think we want to at-least allow duplicate recommissions, or we check that the current service node state has to be different from the one that's going to be added to the blockchain.